### PR TITLE
ShaderLab ignore `Editor` module

### PR DESCRIPTION
**Pull request by Sway007**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)

related editor issue galacean/editor#2877

1. ShaderLab compiler ignore `Editor` module like below 
```glsl
Shader "Unlit" {
 ...
  Editor {
            Properties {
              /**
               * @language zh
               * 主颜色
              */

              /**
              * @language en
              * Main Color
              */
              material_BaseColor("Main Color", Color) = (0, 0, 0, 1);
              material_AlphaCutoff("Alpha Cutoff", Range(0, 1, 0.01)) = 0;
              material_BaseTexture("Texture", Texture2D);
            }

            Macros {
              Header("Conditional Macors") {
                MATERIAL_HAS_BASETEXTURE("Base Texture");
                MATERIAL_IS_ALPHA_CUTOFF("Alpha Cutoff");
              }
            }

            UIScript "${params.uiScriptPath}";
          }
 ...
}
```

2. Fix `RenderQueueType` variable assignment bug in `ShaderLab`
```
// renderQueueType is a variable evaluea
RenderQueueType = renderQueueType;
```

3. Remove duplicate code skip process.

Editor block code skip process is performed in `ShaderContentParser` and `PpParser`

## Release Notes

- **New Features**
	- Introduced the `GS_Editor` keyword to enhance shader processing capabilities.
	- Added new mappings for the `Editor` keyword in the keyword mapping system.

- **Bug Fixes**
	- Improved error handling during shader parsing, ensuring robust logging and management of parsing errors.
	- Updated error reporting mechanisms to enhance clarity and consistency across various components.

- **Tests**
	- Expanded test suite with additional shader validation tests to ensure accuracy in shader processing and error handling.
	- Adjusted expectations for error handling in the release version of shader parsing.

- **Documentation**
	- Updated method signatures to reflect changes in error handling return types for better clarity.

### CODE REVIEW

File: packages/shader-lab/src/preprocessor/PpParser.ts [hunk 1]
```diff
@@ -647,7 +647,7 @@ export class PpParser {
   }
 
   private static _skipEditorBlock(token: BaseToken, scanner: PpScanner) {
-    if (token.lexeme === "EditorProperties" || token.lexeme === "EditorMacros") {
+    if (token.lexeme === "EditorProperties" || token.lexeme === "EditorMacros" || token.lexeme === "Editor") {
```
Comment (zhuxudong) [inline packages/shader-lab/src/preprocessor/PpParser.ts:650]: `["EditorProperties","EditorMacros","Editor"].indexOf(token.lexeme) !== -1`

Comment (Sway007) [reply packages/shader-lab/src/preprocessor/PpParser.ts]: done

File: packages/shader-lab/src/ShaderLabUtils.ts [hunk 2]
```diff
@@ -27,11 +27,12 @@ export class ShaderLabUtils {
     source: string,
     location: ShaderRange | ShaderPosition,
     file?: string
-  ) {
+  ): GSError {
     // #if _VERBOSE
     return new GSError(errorName, message, location, source, file);
     // #else
-    throw new Error(`[${errorName}]: ${message}`);
+    // @ts-ignore
```
Comment (GuoLei1990) [inline packages/shader-lab/src/ShaderLabUtils.ts:34]: type error

File: packages/shader-lab/src/contentParser/KeywordMap.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ export const KeywordMap = new Map([
   ["RasterState", EKeyword.GS_RasterState],
   ["EditorProperties", EKeyword.GS_EditorProperties],
   ["EditorMacros", EKeyword.GS_EditorMacros],
+  ["Editor", EKeyword.GS_Editor],
```
Comment (GuoLei1990) [inline packages/shader-lab/src/contentParser/KeywordMap.ts:11]: Should we ignore this in runtime?

**Pull request merged**